interface/usergroup/facility_user_admin.php

```diff
@@ -140,56 +160,62 @@
                 <input type=hidden name=mode value="facility_user_id">
-                <input type=hidden name=user_id value="<?php echo attr($_GET["user_id"]);?>">
-                <input type=hidden name=fac_id value="<?php echo attr($_GET["fac_id"]);?>">
+                <input type=hidden name=user_id value="<?php echo attr($_GET["user_id"]); ?>">
+                <input type=hidden name=fac_id value="<?php echo attr($_GET["fac_id"]); ?>">
                 <?php $iter = sqlQuery("select * from facility_user_ids where id=?", array($my_id)); ?>
 
                 <table border=0 cellpadding=0 cellspacing=0>
                     <tr>
                         <td>
-                            <?php echo xlt('User'); ?>:
+                            <p><?php echo xlt('User'); ?>:</p>
                         </td>
                         <td>
-                            <?php echo text($user_info['username']); ?>
+                            <p> <?php echo text($user_info['username']); ?> <p>
                         </td>
                     </tr>
                     <tr>
                         <td>
-                            <?php echo xlt('Facility'); ?>:
+                            <p> <?php echo xlt('Facility'); ?>: </p>
                         </td>
                         <td>
-                            <?php echo text($fac_info['name']); ?>
+                            <p> <?php echo text($fac_info['name']); ?> </p>
                         </td>
                     </tr>
                     <?php foreach ($l_arr as $layout_entry) { ?>
                         <tr>
                             <td style="width:180px;">
-                                <?php echo text(xl_layout_label($layout_entry['title'])) ?>:
+                               <p> <?php echo text(xl_layout_label($layout_entry['title'])) ?>: </p>
                             </td>
                             <td style="width:270px;">
                                 <?php
                                 $entry_data = sqlQuery("SELECT `field_value` FROM `facility_user_ids` " .
-                                                       "WHERE `uid` = ? AND `facility_id` = ? AND `field_id` = ?", array($user_info['id'],$fac_info['id'],$layout_entry['field_id']));
-                                echo "<td>" . generate_form_field($layout_entry, $entry_data['field_value']) . "&nbsp;</td>";
+                                    "WHERE `uid` = ? AND `facility_id` = ? AND `field_id` = ?", array($user_info['id'], $fac_info['id'], $layout_entry['field_id']));
+                                if ($layout_entry['field_id'] === "role_code") {
+                                    echo generate_select_list("role", "us-core-provider-role", "", "", " ", "select-dropdown", "", "role");
+                                } else {
+                                    echo generate_form_field($layout_entry, $entry_data['field_value']);
+                                }
```

Comment:
hi @yashrajbothra ,
This is a good first stab at this. note the goal is for this to work with the `generate_form_field($layout_entry, $entry_data['field_value'])` for all cases. The magic is in the data_type setting for the field in layout_options entry. This is where you can choose whether it is a simple text field or a list selector. If a list selector, then the list_id will store the list that is used. When you have a handle on that, then you can create magic by creating a new data_type that basically looks like the list selector with a couple modifications (the select-dropdown class when call generate_select_list from it) to support the search feature. Then this new datatype will be able to be used in any form that uses the layout based engine.

Reply:
Kool :+1:  Didn't new Layout options are so powerful 
Will go ahead and try to make new data type for this case then :+1: